### DISCUSSION TITLE
Zip File upload not working

### DISCUSSION THREAD

**Discussion by thisishasan** [livewire/livewire#3897] [1. Help] [answered]
I got an error when I had tried to upload zip file

the error is 
{"message":"The given data was invalid.","errors":{"files.0":["The files.0 failed to upload."]}}

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/livewire/livewire/discussions/3147

**Comment by jonneroelofs** [comment]
what does your validation look like?

**Reply by thisishasan** [reply]
> what does your validation look like?

 $rules = [
            'file' => ['required', 'file']
        ];

        $this->validate($rules);

this is the code i have written in the component file

**Comment by joshhanley** [comment, marked as answer]
@thisishasan what size zip file were you trying to upload?

If it's bigger than 12MB, you need to make sure to publish the Livewire config and modify the `tempoary_file_upload` rules to allow files bigger than 12MB.

Hope this helps!

**Reply by thisishasan** [reply]
> @thisishasan what size zip file were you trying to upload?
> 
> If it's bigger than 12MB, you need to make sure to publish the Livewire config and modify the `tempoary_file_upload` rules to allow files bigger than 12MB.
> 
> Hope this helps!

thanks the issue is resolved! I have increased value of  upload_max_filesize in php.ini file

**Reply by joshhanley** [reply]
No worries 🙂

**Comment by nexus-4** [comment]
I encountered the exact same issue on macOS while using Laravel Herd Lite. Here is how I resolved it:

Step 1: Locate your active configuration file by running `php --ini`.  
Output example:
```text
Configuration File (php.ini) Path: /usr/local/etc/php
Loaded Configuration File:         /Users/usul/.config/herd-lite/bin/php.ini
```

Step 2: Open the configuration file at `/Users/usul/.config/herd-lite/bin/php.ini` and update the upload limits:
```text
upload_max_filesize = 20M
post_max_size = 20M
```

Step 3: Fix your `config/livewire.php` file by adjusting the temporary file upload rules:
```text
'rules' => 'max:20480', // 20MB in KB
```

Step 4: Clear Laravel's configuration cache:
```text
php artisan config:clear
```

Step 5: Forcefully terminate any lingering PHP processes so they restart with your updated `php.ini` settings:
```text
killall -9 php
```

Note: The final step was essential for me; the earlier steps alone did not resolve the issue properly.
